### DISCUSSION TITLE
Channel creation and resource freeing problems with blocked connection

### DISCUSSION THREAD

**Discussion by telandor** [rabbitmq/rabbitmq-dotnet-client#1255] [General]
I have the following problems when a connection is blocked:
- I can't create any new channels after the connection switches back to running.
- I can't free connection related resources.

See https://stackoverflow.com/questions/73754178/creating-new-channels-after-rabbitmq-connection-was-blocked
and https://stackoverflow.com/questions/73744305/free-resources-in-blocked-rabbitmq-connection

or the Git repository which as well shows a step-by-step procedure to reproduce my issues:
https://github.com/telandor/RabbitMQTest (Edit 26.9.22: made private)

**Comment by michaelklishin** [comment]
You won't be able to do so when an alarm is in effect on connections that have sent at least one publishing frame. RabbitMQ will stop reading from the socket on such connections The easiest solution is to use a separate connection for topology declaration and consumers.

There are also `connection.[un]blocked` events that are sent to clients that support these methods. This one should, too.

If connections were not "unblocked" after all alarms in the cluster clear, we'd likely have this reported almost daily. But that's not the case.

**Reply by telandor** [reply]
Thank you for the quick response.
So, to create a new channel I would need to create a new connection. And freeing resources of a blocked connection is not possible until the connection is 'running' again.

**Reply by michaelklishin** [reply]
You need to create two or so connections early on application boot, then open channels on one for publishers and another for consumers (a connection that never publishes and thus cannot be blocked).

You absolutely [should not open a new connection for every channel](https://www.rabbitmq.com/channels.html).

**Reply by telandor** [reply]
OK, thank you.
